R/PartInv.R

```diff
@@ -89,237 +90,146 @@ NULL
+  argg <- prep_params(argg)
+  out <- new_PartInv()
+  params <- argg[c("weights_item", "weights_latent", "alpha", "psi",
+                   "lambda", "theta", "nu", "pmix", "propsel", "labels",
+                   "cut_z", "num_g")]
+  out$params <- params[
+    c("alpha", "psi", "lambda", "nu", "theta",
+      "weights_item", "weights_latent", "labels", "num_g", "pmix")]
+  out$params[["functioncall"]] <- match.call()
+
+  out_pi <- do.call(compute_cai, params)
+  names_to_update <- intersect(names(out), names(out_pi))
+  out[names_to_update] <- out_pi[names_to_update]
 
   if (out$propsel <= 0.01) warning("Proportion selected is 1% or less.")
 
-  ai_ratio <- as.data.frame(out$summary[5, (num_g + 1):(num_g + num_g - 1)] /
-    out$summary[5, 1])
-
-#    message("Note: The first group is being used as the reference group. 
-#Rearrange the inputs to designate a different group as the reference.\n\n")
-  
-  names(ai_ratio) <- labels[-1]
-  row.names(ai_ratio) <- c("")
-  out[["ai_ratio"]] <- ai_ratio
+  out[["ai_ratio"]] <- get_ai_ratio_partInv(out)
 
   if (show_mi_result) {
-    pop_weights <- pmix
-    lambda_average <- .weighted_average_list(lambda, weights = pop_weights)
-    nu_average <- .weighted_average_list(nu, weights = pop_weights)
-    theta_average <- .weighted_average_list(theta, weights = pop_weights)
-
-    lambda_average_g <- nu_average_g <- theta_average_g <-
-      vector(mode = "list", length = num_g)
+    lambda_avg <- .weighted_average_list(params$lambda, weights = params$pmix)
+    nu_avg <- .weighted_average_list(params$nu, weights = params$pmix)
+    theta_avg <- .weighted_average_list(params$theta, weights = params$pmix)
 
-    for (i in 1:num_g) {
-      lambda_average_g[[i]] <- lambda_average
-      nu_average_g[[i]] <- nu_average
-      theta_average_g[[i]] <- theta_average
-    }
+    params[["lambda"]] <- replicate(params$num_g, lambda_avg, simplify = FALSE)
+    params[["nu"]] <- replicate(params$num_g, nu_avg, simplify = FALSE)
+    params[["theta"]] <- replicate(params$num_g, theta_avg, simplify = FALSE)
 
-    out_mi <- compute_cai(weights_item, weights_latent, alpha, psi,
-      lambda_average_g, nu_average_g, theta_average_g,
-      pmix, propsel, labels, cut_z,
-      is_mi = TRUE
-    )
-    colnames(out_mi$summary) <- labels
+    out_mi <- do.call(compute_cai, c(params, list(is_mi = TRUE)))
     names(out_mi) <- paste0(names(out_mi), "_mi")
-    out <- c(out, out_mi)
-  }
+    names_to_update <- intersect(names(out), names(out_mi))
+    out[names_to_update] <- out_mi[names_to_update]
+    # colnames(out_mi$summary) <- params$labels
+    # names(out_mi) <- paste0(names(out_mi), "_mi")
 
+  }
 
   if (plot_contour) {
-    plot.PartInv(out, labels = labels, which_result = "pi", ...)
     if (show_mi_result == TRUE) {
-      plot.PartInv(out, labels = labels, which_result = "mi", ...)
+      which_result <- c("pi", "mi")
+    } else {
+      which_result <- c("pi")
     }
+    dots <- list(...)
```

Comment:
I'm not following why we need to use `do.call()` in this case, instead of calling `plot.PartInv()` directly. Could you clarify? Thanks.